### PR TITLE
Upgraded Dependencies and Removed Mongoose deprecation warning

### PR DESCRIPTION
**Pull request by nartc**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
- Upgraded dependencies:

1. Mongoose: 5.2.14
2. Typescript 3
3. NestJS Core and Common peer dependencies

- Removed deprecation warning regarding the `useCreateIndex` with mongoose 5.2.9+

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe: Upgrading
```

## What is the current behavior?

Contains deprecation warning.

Issue Number: N/A

## What is the new behavior?
Removed deprecation warning

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: lib/mongoose-core.module.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export class MongooseCoreModule {
       ...mongooseOptions
     } = options;
 
+    mongoose.set('useCreateIndex', true);
```
Comment (fernandolguevara) [inline lib/mongoose-core.module.ts]: useCreateIndex must be an optional settings

File: lib/mongoose-core.module.ts [hunk 2]
```diff
@@ -86,18 +88,11 @@ export class MongooseCoreModule {
           ...mongooseOptions
         } = mongooseModuleOptions;
 
+        mongoose.set('useCreateIndex', true);
```
Comment (fernandolguevara) [inline lib/mongoose-core.module.ts]: useCreateIndex must be an optional settings

File: lib/mongoose-core.module.ts (hunk 1 above)
Comment (kamilmysliwiec) [reply lib/mongoose-core.module.ts]: @nartc why is it required?

Comment (nartc) [reply lib/mongoose-core.module.ts]: It is a deprecation warning on my end. This is a hasty yet careless PR. I'll make change as soon as possible. https://mongoosejs.com/docs/deprecations.html

**Pull request closed without merging**